core/src/main/java/com/rudderstack/android/sdk/core/RudderContext.java

```diff
@@ -63,10 +68,12 @@ private RudderContext() {
         this.libraryInfo = new RudderLibraryInfo();
     }
 
+
     void updateTraits(RudderTraits traits) {
         // if traits is null reset the traits to a new one with only anonymousId
         if (traits == null) {
-            traits = new RudderTraits(this.getDeviceId());
+            RudderPreferenceManager preferenceManger = RudderPreferenceManager.getInstance(RudderClient.getInstance().getApplication());
+            traits = new RudderTraits(preferenceManger.getAnonymousId());
```

Comment:
You can simply call RudderTraits().

Better to remove the constructor in RudderTraits(String anonymousId) ==> it was needed to initialize with a deviceId

Reply:
changed